src/coreclr/hosts/corerun/corerun.cpp

```diff
@@ -1,544 +1,322 @@
+    if (core_root.empty())
     {
-        return true;
+        string_t file;
+        pal::split_path_to_dir_filename(exe_path, core_root, file);
+        pal::ensure_trailing_delimiter(core_root);
     }
-    // Check if a hostpolicy exists and if it does, load it.
-    if (INVALID_FILE_ATTRIBUTES != ::GetFileAttributesW(hostPolicyPath.GetUnicode()))
+    else
     {
-        hMod = ::LoadLibraryExW(hostPolicyPath.GetUnicode(), nullptr, LOAD_LIBRARY_SEARCH_DLL_LOAD_DIR | LOAD_LIBRARY_SEARCH_DEFAULT_DIRS);
+        pal::ensure_trailing_delimiter(core_root);
+        native_search_dirs << core_root << pal::env_path_delim;
     }
 
-    return hMod != nullptr;
-}
+    string_t tpa_list = build_tpa(core_root, core_libs);
 
-HRESULT InitializeHost(
-    Logger& log,
-    HostEnvironment& host,
-    const SString& appPath,
-    const SString& appNiPath,
-    const SString& nativeDllSearchDirs,
-    void **hostHandle,
-    unsigned int *domainId)
-{
-    StackScratchBuffer hostExeNameUTF8;
-    StackScratchBuffer tpaListUTF8;
-    StackScratchBuffer appPathUTF8;
-    StackScratchBuffer appNiPathUTF8;
-    StackScratchBuffer nativeDllSearchDirsUTF8;
+    {
+        // Load hostpolicy if requested.
+        string_t mock_hostpolicy = pal::getenv(envvar::mockHostPolicy);
+        if (!mock_hostpolicy.empty()
+            && !pal::try_load_hostpolicy(mock_hostpolicy))
+        {
+            return -1;
+        }
+    }
 
-    STARTUP_FLAGS flags = CreateStartupFlags();
+    actions.before_coreclr_load();
+
+    // Attempt to load CoreCLR.
+    pal::mod_t coreclr_mod;
+    if (!pal::try_load_coreclr(core_root, coreclr_mod))
+    {
+        return -1;
+    }
 
-    //-------------------------------------------------------------
+    // Get CoreCLR exports
+    coreclr_initialize_ptr coreclr_init_func = nullptr;
+    coreclr_execute_assembly_ptr coreclr_execute_func = nullptr;
+    coreclr_shutdown_2_ptr coreclr_shutdown2_func = nullptr;
+    if (!try_get_export(coreclr_mod, "coreclr_initialize", (void**)&coreclr_init_func)
+        || !try_get_export(coreclr_mod, "coreclr_execute_assembly", (void**)&coreclr_execute_func)
+        || !try_get_export(coreclr_mod, "coreclr_shutdown_2", (void**)&coreclr_shutdown2_func))
+    {
+        return -1;
+    }
 
-    // Initialize host.
+    // Construct CoreCLR properties.
+    pal::string_utf8_t tpa_list_utf8 = pal::convert_to_utf8(std::move(tpa_list));
+    pal::string_utf8_t app_path_utf8 = pal::convert_to_utf8(std::move(app_path));
+    pal::string_utf8_t app_path_ni_utf8 = pal::convert_to_utf8(std::move(app_path_ni));
+    pal::string_utf8_t native_search_dirs_utf8 = pal::convert_to_utf8(native_search_dirs.str());
+    const char* enable_server_gc = get_envvar_as_boolean(envvar::serverGc);
+    const char* enable_concurrent_gc = get_envvar_as_boolean(envvar::concurrentGc, true /* default */);
+    const char* enable_globalization_invariant = get_envvar_as_boolean(envvar::globalizationInvariant);
```

Comment:
We have `DOTNET_SYSTEM_GLOBALIZATION_INVARIANT` environment variable that is read and respected by the runtime. I do not think we need another one here.

Reply:
This was folded in from the unix variant. I will see what sets `CORECLR_GLOBAL_INVARIANT` and convert it to that.

Reply:
This looks that a no op. I don't see it consumed anywhere. Will remove.